pkg/api/payer/nodeSelector.go

```diff
@@ -53,9 +54,19 @@ func (s *StableHashingNodeSelectorAlgorithm) GetNode(
 	spacing := maxHashSpace / numNodes
 
 	// Compute virtual positions for each node
+	// Skip nodes that are disabled or do not have an API enabled
 	nodeLocations := make([]uint32, numNodes)
```

Comment:
If this is the length of `numNodes` but some are being skipped in the loop below the array is going to have empty values.

Better to use `append` and calculate `numNodes` based on the filtered list.